### PR TITLE
Additions to cloudbackup enumerate api

### PR DESCRIPTION
**Pull request by talakad**
Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:
Allows users to identify cloudbackups that do not belong to this cluster or whose source volume may already be deleted.
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -398,8 +398,16 @@ type CloudBackupGenericRequest struct {
 	// MetadataFilter indicates backups whose metadata has these kv pairs
 	MetadataFilter map[string]string
 	// CloudBackupID must be specified if one needs to enumerate known single
-	// backup( format is clusteruuidORBucketName/srcVolId-SnapId(-incr)
+	// backup( format is clusteruuidORBucketName/srcVolId-SnapId(-incr). If t\
```
Comment (sangleganesh) [inline api/api.go:401]: nit: space after backup and no space after '('.

// backup (format is clusteruuidORBucketName/SrcVolumeID-SnapId(-incr).
// If specified everything else in the request is ignored

File: api/api.go [hunk 2]
```diff
@@ -398,8 +398,16 @@ type CloudBackupGenericRequest struct {
...
+	// this is specified, everything else n the command is ignored
 	CloudBackupID string
+	// MissingSrcVol set to true enumerates cloudbackups for which srcVol is not
```
Comment (sangleganesh) [inline api/api.go:404]: nit: prefer to use full names when possible.
s/srcVol/SrcVolumeID/

Redo the comment ? The '(' does not end ?

File: api/api.go [hunk 3]
```diff
@@ -416,6 +424,11 @@ type CloudBackupInfo struct {
 	Metadata map[string]string
 	// Status indicates the status of the backup
 	Status string
+	// Cluster indicates if the cloudbackup was uploaded by this
+	// cluster. Could be unknown with older version cloudbackups
+	Cluster SdkCloudBackupClusterType
+	// Namespace to which this cloudbackup belongs to
```
Comment (sangleganesh) [inline api/api.go:430]: is the namespace of the source volume ? 
Also, I would name it ClusterType instead of Cluster for clarity.

**Review by sangleganesh — APPROVED** [review]
minor nits.

File: api/api.go (hunk 3 above)
Comment (talakad) [reply api/api.go:430]: yes. k8s namespace that source volume belongs to.

File: api/api.go (hunk 1 above)
Comment (talakad) [reply api/api.go:401]: Done

File: api/api.go (hunk 2 above)
Comment (talakad) [reply api/api.go:404]: Since the field does not represent ID changed it to MissingSrcVolumes

**Pull request merged**